salt/master.sls

```diff
@@ -2,7 +2,7 @@
 {%- from tplroot ~ "/map.jinja" import salt_settings with context %}
 {%- from tplroot ~ "/libtofs.jinja" import files_switch with context %}
 
-{% if salt_settings.pin_version and salt_settings.version and grains.os_family|lower == 'debian' %}
+{% if salt_settings.install_packages and salt_settings.version and salt_settings.pin_version %}
```

Comment:
Order of conditions does matter, if first condition is `false` and logic operations are `and` Python will not evaluate other conditions. And prior your changes existing order does perfect sense.

I suggest you to install `yum-plugin-versionlock` in separate state file something like `prepare.sls`